### PR TITLE
feat(lib, tool): Added casing options for generated file names

### PR DESCRIPTION
**Pull request by klovaaxel**
Haven't tested it but have written some tests 😅

### CODE REVIEW

File: TypeContractor.Tool/Program.cs [hunk 1]
```diff
@@ -20,6 +20,7 @@
 var buildZodSchemasOptions = new Option<bool>("--build-zod-schemas", () => false, "Enable experimental support for Zod schemas alongside generated types.");
 var generateApiClientsOptions = new Option<bool>("--generate-api-clients", () => false, "Enable experimental support for auto-generating API clients for each endpoint.");
 var apiClientsTemplateOptions = new Option<string>("--api-client-template", () => "aurelia", "Template to use for API clients. Either 'aurelia', 'react-axios' (built-in) or a path to a Handlebars file, including extension");
+var casingOptions = new Option<string>("--casing", () => "pascal", "Casing to use for generated types. Any of 'pascal' (default), 'camel', 'snake', or 'kebab'");
```
Comment (PerfectlyNormal) [inline TypeContractor.Tool/Program.cs:23]: Maybe create an enum (like LogLevel or CleanMethod) to get tab completion and help with documentation

File: TypeContractor/README.md
Comment (PerfectlyNormal) [inline TypeContractor/README.md:1]: ...why do we have a copy of the entire README here? ^^

**Review by PerfectlyNormal — COMMENTED** [review]
Looks nice, but an enum for the casing options would be preferred

File: TypeContractor/README.md
Comment (klovaaxel) [reply TypeContractor/README.md:1]: Yea... I honestly don't know how I have done that...

File: TypeContractor.Tool/Program.cs (hunk 1 above)
Comment (klovaaxel) [reply TypeContractor.Tool/Program.cs:23]: Yeah, I started with an enum but I couldn't figure out how to add an enum so I can access it inside both the Tool and Lib projects, should I have two enums or should I do some fancy namespace thingy? (I mostly tried to copy what you have in CleanMethod but I don't think that is used in other projects)

File: TypeContractor/README.md
Comment (klovaaxel) [reply TypeContractor/README.md:1]: Well, now it's gone 😅

**Pull request merged**